README.md

```diff
@@ -1,2 +1,67 @@
-# thrift-pool
+# Thrift Pool
+
 A module that wraps thrift interfaces in connection pooling logic to make them more resilient.
+
+The [node thrift code](https://www.npmjs.com/package/thrift) exposes a `service` and `types` file.
+The node thrift client also exposes interfaces to create connections to the thrift server.
+There is no way to use the node thrift library to do connection pooling or to create new connections if existing connections fail.
+
+This library takes in a thrift `service` and wraps the methods with connection pooling logic (based on [node-pool](https://github.com/coopernurse/node-pool)).
+
+
+## Installation
+
+```
+npm install node-thrift-pool
+```
+
+## Usage
+```javascript
+var thrift = require('thrift'),
+  Service = require('./gen-nodejs/Service'),
+  Types = require('./gen-nodejs/types'),
+  thriftPool = require('node-thrift-pool');
+
+var thrift_client = thriftPool(thrift, Service, {host: "localhost", port: 9090});
+
+/*
+ * thrift_client is now an initialized thrift client that uses connection pooling behind the scenes
+ * thrift_client.method(function(err, returned_data){console.log(err, returned_data)});
+ */
+
+```
+
+## Supported pooling options
+
+- `host` - **Required** - The host of the thrift server to connect to
+- `port` - **Required** - The port of the thrift server to connect to
+- `max_connections` - Default: `2` - Max number of connections to keep open
```

Comment:
d45898462b795696cc608a75fc25dbfe0688c349

I changed default to 1... it's very easy to change, but since I have no good intuition for what the right max default should be, figured sticking with whatever generic-pool chose as default at least promotes some consistency.   Let me know if you think otherwise.